libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/LayoutSyncManager.java

```diff
@@ -0,0 +1,288 @@
+     * @param communityId Community ID. Pass null if not applicable.
+     * @param smartStore SmartStore instance. Pass null to use current user default SmartStore.
+     * @return Instance of this class.
+     */
+    public static synchronized LayoutSyncManager getInstance(UserAccount account, String communityId,
+                                                             SmartStore smartStore) {
+        if (account == null) {
+            account = SmartSyncSDKManager.getInstance().getUserAccountManager().getCurrentUser();
+        }
+        if (smartStore == null) {
+            smartStore = SmartSyncSDKManager.getInstance().getSmartStore(account, communityId);
+        }
+        final SyncManager syncManager = SyncManager.getInstance(account, communityId, smartStore);
+        final String uniqueId = (account != null ? account.getUserId() : "") + ":"
+                + smartStore.getDatabase().getPath();
+        LayoutSyncManager instance = INSTANCES.get(uniqueId);
+        if (instance == null) {
+            instance = new LayoutSyncManager(smartStore, syncManager);
+            INSTANCES.put(uniqueId, instance);
+        }
+        SalesforceSDKManager.getInstance().registerUsedAppFeature(FEATURE_LAYOUT_SYNC);
+        return instance;
+    }
+
+    /**
+     * Resets all the layout sync managers.
+     */
+    public static synchronized void reset() {
+        INSTANCES.clear();
+    }
+
+    /**
+     * Resets the layout sync manager for this user account.
+     *
+     * @param account User account.
+     */
+    public static synchronized void reset(UserAccount account) {
+        if (account != null) {
+            final Set<String> keysToRemove = new HashSet<>();
+            for (String key : INSTANCES.keySet()) {
+                if (key.startsWith(account.getUserId())) {
+                    keysToRemove.add(key);
+                }
+            }
+            INSTANCES.keySet().removeAll(keysToRemove);
+        }
+    }
+
+    /**
+     * Returns the SmartStore instance used by this instance of LayoutSyncManager.
+     *
+     * @return SmartStore instance.
+     */
+    public SmartStore getSmartStore() {
+        return smartStore;
+    }
+
+    /**
+     * Returns the SyncManager instance used by this instance of LayoutSyncManager.
+     *
+     * @return SyncManager instance.
+     */
+    public SyncManager getSyncManager() {
+        return syncManager;
+    }
+
+    /**
+     * Fetches layout data for the specified object type and layout type using the specified
+     * mode and triggers the supplied callback once complete.
+     *
+     * @param objectType Object type.
+     * @param layoutType Layout type. Defaults to "Full" if null is passed in.
+     * @param mode Fetch mode. See {@link Mode} for available modes.
+     * @param syncCallback Layout sync callback.
+     */
+    public void fetchLayout(String objectType, String layoutType, Mode mode,
```

Comment:
Public API that allows the caller to specify where to fetch data from based on cache policy passed in.